### PR TITLE
fix: refactor config

### PR DESCRIPTION
**Pull request by simboonlong**

### CODE REVIEW

File: src/implementations/config/create.ts [hunk 1]
```diff
@@ -1,98 +1,143 @@
...
 
-  const getContractAddress = async (
-    typesOfForms: TypesOfForms[],
-    formType: "VERIFIABLE_DOCUMENT" | "TRANSFERABLE_RECORD",
-    identityProofType: "DNS-DID" | "DNS-TXT"
-  ): Promise<string> => {
-    const isValidForm = typesOfForms.some(
-      (item: TypesOfForms) =>
-        item.type === formType && (<any>Object).values(item.identityProofTypes).includes(identityProofType)
-    );
```
Comment (isaackps) [inline src/implementations/config/create.ts:37]: need revert back

**Review by isaackps — CHANGES_REQUESTED** [review]

File: src/implementations/config/create.ts (hunk 1 above)
Comment (simboonlong) [reply src/implementations/config/create.ts:37]: reverted back validate test

File: src/commands/config/create.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export const command = "create [options]";
 export const describe = "Create a config file";
 
 export const builder = (yargs: Argv): Argv =>
-  withWalletOption(
+  withNetworkAndWalletSignerOption(
```
Comment (isaackps) [inline src/commands/config/create.ts:17]: can revert back to withWalletOption

File: src/implementations/config/types.ts [hunk 3]
```diff
@@ -1,24 +1,37 @@
...
-  defaults: OpenAttestationDocument;
-}
+type WalletAwsKms = {
+  type: string;
+  accessKeyId: string;
+  region: string;
+  kmsKeyId: string;
+};
+
+type Wallet = WalletEncrypted | WalletAwsKms;
```
[the remark this answers is not in the document]
Comment (simboonlong) [reply src/implementations/config/types.ts]: yeah ok

File: src/commands/config/create.ts (hunk 2 above)
Comment (simboonlong) [reply src/commands/config/create.ts]: ok

File: src/implementations/config/create.ts [hunk 4]
```diff
@@ -1,98 +1,160 @@
...
-    return {
-      type: form.type,
-      identityProofTypes: identityProofTypes,
-    };
-  });
+  let tokenRegistryAddress = "";
+  let documentStoreAddress = "";
+  let dnsNameTransferableRecord: string | undefined = "";
+  let dnsNameVerifiable: string | undefined = "";
+  let dnsNameDid: string | undefined = "";
```
Comment (isaackps) [inline src/implementations/config/create.ts:95]: in this case since its an empty string already do we need the undefined?

**Review by isaackps — APPROVED** [review]

**Pull request merged**